### PR TITLE
MAINT: random: Match type of SeedSequence.pool_size to DEFAULT_POOL_SIZE.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
Fixes this compiler warning:

gcc: numpy/random/bit_generator.c
numpy/random/bit_generator.c:4889:41: warning: comparison of integers of different signs: 'int' and 'uint32_t' (aka 'unsigned int') [-Wsign-compare]
  __pyx_t_6 = ((__pyx_v_self->pool_size != __pyx_v_5numpy_6random_13bit_generator_DEFAULT_POOL_SIZE) != 0);
                ~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated

<!-- Please be sure you are following the instructions in the dev guidelines
http://www.numpy.org/devdocs/dev/development_workflow.html
-->

<!-- We'd appreciate it if your commit message is properly formatted
http://www.numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message
-->

### CODE REVIEW

File: numpy/random/bit_generator.pxd [hunk 1]
```diff
@@ -14,7 +14,7 @@ cdef class BitGenerator():
 cdef class SeedSequence():
     cdef readonly object entropy
     cdef readonly tuple spawn_key
-    cdef readonly int pool_size
+    cdef readonly uint32_t pool_size
```
Comment (eric-wieser) [inline numpy/random/bit_generator.pxd]: Is there a reason this (and the other counts in this struct) isn't size_t?

Comment (rkern) [reply numpy/random/bit_generator.pxd]: `pool_size` is supposed to be tiny (relatively speaking). `n_children_spawned` could be largish, but should be capped at `uint32_t` (as it turns into a `uint32_t` element in the `spawn_key`).

Comment (WarrenWeckesser) [reply numpy/random/bit_generator.pxd]: The docstring for the `pool_size` attribute says "Size of the pooled entropy to store. Default is 4 to give a 128-bit entropy pool. 8 (for 256 bits) is another reasonable choice if working with larger PRNGs, but there is very little to be gained by selecting another value."  So 32 bits is far more than necessary, but I guess it is harmless to leave it that size.

>  `n_children_spawned` could be largish, but should be capped at `uint32_t` (as it turns into a `uint32_t` element in the `spawn_key`).

OK, I changed `n_children_spawned` to `uint32_t` in the changes that I just pushed.

**Comment by WarrenWeckesser** [comment]
In the latest update, both `SeedSequence.pool_size` and `SeedSequence.n_children_spawned` are changed to `uint32_t`.

**Review by charris — APPROVED** [review]

File: numpy/random/bit_generator.pyx [hunk 2]
```diff
@@ -116,7 +116,7 @@ def _coerce_to_uint32_array(x):
     Examples
     --------
     >>> import numpy as np
-    >>> from np.random.bit_generator import _coerce_to_uint32_array
+    >>> from numpy.random.bit_generator import _coerce_to_uint32_array
```
Comment (mattip) [inline numpy/random/bit_generator.pyx]: Why is this needed? The line above should make the old code work ...

Comment (WarrenWeckesser) [reply numpy/random/bit_generator.pyx]: > Why is this needed?

I'm not sure which part of the change "this" refers to.  The import statement is needed so we can use `_coerce_to_uint32_array` without giving the fully qualified name.  The change from `np` to `numpy` is needed because a Python variable can't be used as part of the module in an import statement.

Comment (charris) [reply numpy/random/bit_generator.pyx]: I don't think the `import numpy as np` is needed for examples, but that is extraneous to this PR.

**Comment by charris** [comment]
@WarrenWeckesser Some lines in your commit message are way long. Might want to check your editor setting for them. Most editors have plugins for git commit messages.

**Comment by WarrenWeckesser** [comment]
@charris, the long lines are from the compiler warning that I included in the commit message.  I intentionally didn't break that up, but I can do so in the future.
